packages/web/src/common/store/cache/tracks/utils/reformat.ts

```diff
@@ -61,6 +66,31 @@ const setDefaultFolloweeSaves = <T extends TrackMetadata>(track: T) => {
   }
 }
 
+const setTypedPremiumConditions = <T extends TrackMetadata>(track: T) => {
+  const premium_conditions = track.premium_conditions as any
+  let type
+  if (premium_conditions?.nft_collection != null) {
+    type = PremiumContentType.COLLECTIBLE_GATED
+  } else if (premium_conditions?.follow_user_id != null) {
+    type = PremiumContentType.FOLLOW_GATED
+  } else if (premium_conditions?.tip_user_id != null) {
+    type = PremiumContentType.TIP_GATED
+  } else if (premium_conditions?.usdc_purchase != null) {
+    type = PremiumContentType.USDC_PURCHASE
+  } else {
+    type = null
```

Comment:
Oof, ideally we shouldn't need to do this.
I wonder if it's worth updating `PremiumConditionsBase` to a _different_ discriminated union where at least one of the nested properties is guaranteed to exist if the object exists at all? 

```
type PremiumConditionsBase = 
 | {
      nft_collection:
      | PremiumConditionsEthNFTCollection
      | PremiumConditionsSolNFTCollection 
  } 
  | { follow_user_id: number }
  | { tip_user_id: number }
  | { usdc_purchase: PremiumConditionsUSDCPurchase }
```
Essentially we're saying "if this conditions object exists at all, one and only one of the fields will have a value".

Is that a crazy idea? 😅

Reply:
I think this is the right idea but it's giving me a million type errors, the consumers aren't set up on those assumptions. Since we're trying to migrate away from the old world anyway I think this is probably fine? The `PremiumConditionsBase` needs to just match exactly what the old world had.

Reply:
Trying to think of a better way. Since `type` doesn't come from the backend and we're using `as any` it's hard to guarantee that at least one of the fields exist?

Reply:
My 2 cents: I don't mind the null check here since it's guaranteeing we get either a properly typed premium_conditions or none at all returned, due to the conditional